test/uk/gov/hmrc/iossreturns/controllers/ReturnStatusControllerSpec.scala

```diff
@@ -105,17 +107,22 @@ class ReturnStatusControllerSpec
 
           status(result) mustEqual OK
           contentAsJson(result) mustEqual Json.toJson(CurrentReturns(Seq(
-          ), excluded = false, finalReturnsCompleted = false, iossNumber))
+          ), excluded = false, finalReturnsCompleted = false, iossNumber,
+            Seq.empty))
         }
       }
 
       "with no returns in progress, due or overdue if all returns are complete" in {
-
         val mockReturnService = mock[ReturnsService]
         val lastPeriod = periods.takeRight(1).head
-        when(mockReturnService.getStatuses(any(), any(), any())) thenReturn Future.successful(
-          periods.dropRight(1).map(PeriodWithStatus(_, Complete)).toList ::: List(PeriodWithStatus(lastPeriod, Next))
-        )
+        val periodsWithStatuses = periods.dropRight(1).map(PeriodWithStatus(_, Complete)).toList ::: List(PeriodWithStatus(lastPeriod, Next))
+
+        val completeOrExcludedReturns: List[Return] = convertPeriodsWithStatusesToCompleteOrExcludedReturns(periodsWithStatuses)
+
+        // Check we are not doing an empty list and in fact negating the value of the test, stranger things have happened
+        completeOrExcludedReturns.length mustBe 14
```

Comment:
`stranger things have happened` what happened? :)

Reply:
There was an empty list being compared to an empty list when what was expected was not an empty list on either side of the check. Was to do with security logic for who could see which medical research data. The user should have seen some data, but the setup was so complicated it wasn't possible to spot that the test was actually another no-data one.